component/service/src/main/java/org/exoplatform/social/gettingStarted/service/GettingStartedService.java

```diff
@@ -0,0 +1,144 @@
+package org.exoplatform.social.gettingStarted.service;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Locale;
+import java.util.ResourceBundle;
+
+import org.exoplatform.commons.utils.CommonsUtils;
+import org.exoplatform.commons.utils.ListAccess;
+import org.exoplatform.container.ExoContainerContext;
+import org.exoplatform.services.log.ExoLogger;
+import org.exoplatform.services.log.Log;
+import org.exoplatform.services.resources.ResourceBundleService;
+import org.exoplatform.services.security.ConversationState;
+import org.exoplatform.social.common.RealtimeListAccess;
+import org.exoplatform.social.core.identity.model.Identity;
+import org.exoplatform.social.core.identity.model.Profile;
+import org.exoplatform.social.core.identity.provider.OrganizationIdentityProvider;
+import org.exoplatform.social.core.manager.ActivityManager;
+import org.exoplatform.social.core.manager.IdentityManager;
+import org.exoplatform.social.core.manager.RelationshipManager;
+import org.exoplatform.social.core.space.model.Space;
+import org.exoplatform.social.core.space.spi.SpaceService;
+import org.exoplatform.social.gettingStarted.dto.GettingStartedStep;
+
+public class GettingStartedService {
+
+  private static final Log         LOG = ExoLogger.getLogger(GettingStartedService.class);
+
+  private List<GettingStartedStep> gettingStartedSteps;
+
+  public List<GettingStartedStep> getUserSteps(String lang) {
+
+    String userId = ConversationState.getCurrent().getIdentity().getUserId();
+    List<GettingStartedStep> gettingStartedSteps = new ArrayList<>();
+    ResourceBundleService resourceBundleService = CommonsUtils.getService(ResourceBundleService.class);
+    ResourceBundle resourceBundle = resourceBundleService.getResourceBundle("locale.portlet.social.GettingStartedPortlet",
+                                                                            new Locale(lang));
+
+    GettingStartedStep stepAvatar = new GettingStartedStep();
+    stepAvatar.setName(resourceBundle.getString("social.portlet.gettingStarted.avatar"));
```

Comment:
I think that it's better to let I18N in portlet only. Here, we can return a keyword ('avatar' by example)
And in portlet, you use:
``
$t(`social.portlet.gettingStarted.${step.name}`)
``
In addition, in I18N file, please use 'locale.portlet.GettingStartedPortlet' (without 'social'). Same thing for keywords (`gettingStarted.step.avatar` instead of `social.portlet.gettingStarted.avatar`)

Reply:
Done!